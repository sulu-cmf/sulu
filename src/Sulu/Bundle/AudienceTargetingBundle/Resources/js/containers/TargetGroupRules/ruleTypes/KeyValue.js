// @flow
import React from 'react';
import {Input} from 'sulu-admin-bundle/components';
import type {RuleTypeProps} from '../types';
import keyValueStyles from './keyValue.scss';

export default class KeyValue extends React.Component<RuleTypeProps> {
    handleParameterChange = (parameter: ?string) => {
        const {onChange, options} = this.props;
        onChange({...this.props.value, [options.keyName]: parameter});
    };

    handleValueChange = (value: ?string) => {
        const {onChange, options} = this.props;
        onChange({...this.props.value, [options.valueName]: value});
    };

    render() {
        const {options, value} = this.props;
        const {keyName, keyPlaceholder, valueName, valuePlaceholder} = options;

        return (
            <div className={keyValueStyles.inputs}>
                <Input onChange={this.handleParameterChange} placeholder={keyPlaceholder} value={value[keyName]} />
                <Input onChange={this.handleValueChange} placeholder={valuePlaceholder} value={value[valueName]} />
            </div>
        );
    }
}
