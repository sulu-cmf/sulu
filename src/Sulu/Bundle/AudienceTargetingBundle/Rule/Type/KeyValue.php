<?php

/*
 * This file is part of Sulu.
 *
 * (c) Sulu GmbH
 *
 * This source file is subject to the MIT license that is bundled
 * with this source code in the file LICENSE.
 */

namespace Sulu\Bundle\AudienceTargetingBundle\Rule\Type;

class KeyValue implements RuleTypeInterface
{
    /**
     * @var string
     */
    private $keyName;

    /**
     * @var string
     */
    private $valueName;

    /**
     * @var string
     */
    private $keyPlaceholder;

    /**
     * @var string
     */
    private $valuePlaceholder;

    public function __construct($keyName, $valueName, $keyPlaceholder, $valuePlaceHolder)
    {
        $this->keyName = $keyName;
        $this->valueName = $valueName;
        $this->keyPlaceholder = $keyPlaceholder;
        $this->valuePlaceholder = $valuePlaceHolder;
    }

    public function getName(): string
    {
        return 'key_value';
    }

    public function getOptions(): array
    {
        return [
            'keyName' => $this->keyName,
            'valueName' => $this->valueName,
            'keyPlaceholder' => $this->keyPlaceholder,
            'valuePlaceholder' => $this->valuePlaceholder,
        ];
    }
}
