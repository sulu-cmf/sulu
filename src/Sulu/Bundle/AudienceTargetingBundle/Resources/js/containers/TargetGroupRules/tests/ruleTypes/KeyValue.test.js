// @flow
import React from 'react';
import {render, shallow} from 'enzyme';
import KeyValue from '../../ruleTypes/KeyValue';

test('Render a KeyValue RuleType', () => {
    const options = {
        keyPlaceholder: 'key',
        valuePlaceholder: 'value',
    };

    expect(render(<KeyValue onChange={jest.fn()} options={options} value={{}} />)).toMatchSnapshot();
});

test.each([
    ['test1', 'value1', {}, {test1: 'value1'}],
    ['test2', 'value2', {test2: 'value1'}, {test2: 'value2'}],
    ['test2', 'value2', {test1: 'value1'}, {test1: 'value1', test2: 'value2'}],
])('Call onChange handler when value is changed for "%s" to "%s"', (valueName, value, oldValue, result) => {
    const changeSpy = jest.fn();

    const options = {
        valueName,
    };

    const keyValue = shallow(<KeyValue onChange={changeSpy} options={options} value={oldValue} />);
    keyValue.find('Input').at(1).prop('onChange')(value);

    expect(changeSpy).toBeCalledWith(result);
});

test.each([
    ['test1', 'key1', {}, {test1: 'key1'}],
    ['test2', 'key2', {test2: 'key1'}, {test2: 'key2'}],
    ['test2', 'key2', {test1: 'key1'}, {test1: 'key1', test2: 'key2'}],
])('Call onChange handler when key is changed for "%s" to "%s"', (keyName, key, oldValue, result) => {
    const changeSpy = jest.fn();

    const options = {
        keyName,
    };

    const keyValue = shallow(<KeyValue onChange={changeSpy} options={options} value={oldValue} />);
    keyValue.find('Input').at(0).prop('onChange')(key);

    expect(changeSpy).toBeCalledWith(result);
});
