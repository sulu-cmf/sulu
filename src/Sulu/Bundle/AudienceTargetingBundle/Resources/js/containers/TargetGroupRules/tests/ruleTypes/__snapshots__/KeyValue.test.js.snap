// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a KeyValue RuleType 1`] = `
<div
  class="inputs"
>
  <label
    class="input default left"
  >
    <input
      placeholder="key"
      type="text"
      value=""
    />
  </label>
  <label
    class="input default left"
  >
    <input
      placeholder="value"
      type="text"
      value=""
    />
  </label>
</div>
`;
